### DISCUSSION TITLE
Podman: cannot set up namespace using "/usr/bin/newuidmap"

### DISCUSSION THREAD

**Discussion by tarun360** [containers/podman#17569] [Q&A]
I faced a transient error with podman, as follows:

```
$ podman images
ERRO[0000] running `/usr/bin/newuidmap 126747 0 21363 1 1 2000000 65536 65537 2000000 65536 131073 2000000 65536 196609 2000000 65536 262145 1334579568 65536`: newuidmap: write to uid_map failed: Invalid argument 
Error: cannot set up namespace using "/usr/bin/newuidmap": exit status 1
$ podman pull some_remote_image
ERRO[0000] running `/usr/bin/newuidmap 126694 0 21363 1 1 2000000 65536 65537 2000000 65536 131073 2000000 65536 196609 2000000 65536 262145 1334579568 65536`: newuidmap: write to uid_map failed: Invalid argument 
Error: cannot set up namespace using "/usr/bin/newuidmap": exit status 1
```

I was able to get around this error by specifying `--cgroup-manager=cgroupfs`, as follows:
```
$ podman --cgroup-manager cgroupfs images
REPOSITORY  TAG         IMAGE ID    CREATED     SIZE
```

After specifying the cgroup manager as cgroupfs once (as above), I now no longer get the above errors, whether I specify cgroup manager as systemd or as cgroupfs.
```
$ podman --cgroup-manager systemd images
REPOSITORY  TAG         IMAGE ID    CREATED     SIZE
$ podman --cgroup-manager cgroupfs images
REPOSITORY  TAG         IMAGE ID    CREATED     SIZE
```

Since I am unable to reproduce the above transient error, I am unable to find the root cause for this. Does anyone have an idea about the cause of the above error, and how I can prevent it in future?

**Comment by vrothberg** [comment]
Thanks for reachign out, @tarun360!
@giuseppe PTAL

**Comment by giuseppe** [comment]
how does the configuration in `/etc/subuid` and `/etc/subgid` look like?

Is your user added there as part of the assigned ranges?

**Reply by tarun360** [reply]
Yes. My uid is `21363`, so I get the following range for my uid:
```
$ cat /etc/subuid | grep 21363
21363:1334579568:65536
```

**Reply by tarun360** [reply]
Hi @giuseppe , were you able to find anything on this?

Thanks,
-Tarun

**Reply by giuseppe** [reply]
that looks correct.

Can you check if newuidmap has capabilities?

```
$ getfattr -d -m'.*' /usr/bin/newuidmap  /usr/bin/newgidmap 
getfattr: Removing leading '/' from absolute path names
# file: usr/bin/newuidmap
security.capability=0sAQAAAoAAAAAAAAAAAAAAAAAAAAA=
security.selinux="system_u:object_r:bin_t:s0"

# file: usr/bin/newgidmap
security.capability=0sAQAAAkAAAAAAAAAAAAAAAAAAAAA=
security.selinux="system_u:object_r:bin_t:s0"
```

**Reply by tarun360** [reply]
It does have:
```
$ getfattr -d -m'.*' /usr/bin/newuidmap  /usr/bin/newgidmap 
getfattr: Removing leading '/' from absolute path names
# file: usr/bin/newuidmap
security.capability=0sAQAAAoAAAAAAAAAAAAAAAAAAAAA=

# file: usr/bin/newgidmap
security.capability=0sAQAAAkAAAAAAAAAAAAAAAAAAAAA=
```

**Reply by giuseppe** [reply]
can you try running the following commands?

```
$ unshare -U sleep 100 &
$ newuidmap $! 0 1334579568 65536
```

**Reply by tarun360** [reply]
```
$ unshare -U sleep 100 &
[1] 1259718
$ newuidmap $! 0 1334579568 65536
```
(There's no output of the second command)

**Reply by giuseppe** [reply]
can you please show the full content of `/etc/subuid`?

I am not sure it contains only what you've shown above since Podman is attempting to use `0 21363 1 1 2000000 65536 65537 2000000 65536 131073 2000000 65536 196609 2000000 65536 262145 1334579568 65536`

so the `2000000` initial range is used multiple times.

**Reply by tarun360** [reply]
> can you please show the full content of /etc/subuid?

Previously I showed you output of the command `cat /etc/subuid | grep 21363` only. There are lot rows in `/etc/subuid`, do you want me to grep a particular string in it?

**Reply by giuseppe** [reply]
show the entire file

**Reply by giuseppe** [reply]
is your user name also present there (not just the UID)?

**Reply by tarun360** [reply]
My username is not present there, just the UID is present.

> show the entire file

Apologies, but I can't share the entire file due to regulatory reasons. If you want me to grep something, I can help with that.

**Reply by giuseppe** [reply]
are you sure your system is configured to use `/etc/subuid` and not IPA (https://freeipa.readthedocs.io/en/latest/designs/subordinate-ids.html)?

**Reply by tarun360** [reply]
Could you please tell how can I check that?

**Reply by giuseppe** [reply]
use the `getsubids` tool.

`getsubids $YOUR_USER_NAME`
